test/controllers/incomeSources/add/UKPropertyAddedControllerSpec.scala

```diff
@@ -0,0 +1,166 @@
+
+  val view: UKPropertyAdded = app.injector.instanceOf[UKPropertyAdded]
+  val mockDateService: DateService = mock(classOf[DateService])
+
+  object TestUKPropertyAddedController extends UKPropertyAddedController(
+    MockAuthenticationPredicate,
+    mockAuthService,
+    app.injector.instanceOf[SessionTimeoutPredicate],
+    mockDateService,
+    mockIncomeSourceDetailsService,
+    mockNextUpdatesService,
+    app.injector.instanceOf[NavBarPredicate],
+    MockIncomeSourceDetailsPredicate,
+    app.injector.instanceOf[NinoPredicate],
+    view
+  )(appConfig,
+    mcc = app.injector.instanceOf[MessagesControllerComponents],
+    ec, app.injector.instanceOf[ItvcErrorHandler],
+    app.injector.instanceOf[AgentItvcErrorHandler])
+
+  object UKPropertyAddedTestConstants {
+  }
+
+  "UKPropertyAddedController.getBackUrl()" should {
+    "return the correct back URL for individual user" in {
+      val expectedBackUrl = controllers.incomeSources.add.routes.UKPropertyReportingMethodController.show(testSelfEmploymentId).url
+      val backUrl = TestUKPropertyAddedController.getBackUrl(testSelfEmploymentId, isAgent = false)
+
+      backUrl shouldBe expectedBackUrl
+    }
+    "return the correct back URL for agent user" in {
+      val expectedBackUrl = controllers.incomeSources.add.routes.UKPropertyReportingMethodController.showAgent(testSelfEmploymentId).url
+      val backUrl = TestUKPropertyAddedController.getBackUrl(testSelfEmploymentId, isAgent = true)
+
+      backUrl shouldBe expectedBackUrl
+    }
+  }
+
+  "UKPropertyAddedController.show" should {
+    "return 200 OK" when {
+      "FS enabled with newly added UK Property and obligations view model" in {
+        disable(NavBarFs)
+        enable(IncomeSources)
+        setupMockAuthRetrievalSuccess(BaseTestConstants.testIndividualAuthSuccessWithSaUtrResponse())
+        mockUKPropertyIncomeSource()
+
+
+        when(mockDateService.getCurrentTaxYearStart(any())).thenReturn(LocalDate.of(2023, 4, 6))
+
+        when(mockNextUpdatesService.getObligationsViewModel(any(), any())(any(), any(), any())).thenReturn(
+          Future(IncomeSourcesObligationsTestConstants.viewModel))
+
+        when(mockNextUpdatesService.getNextUpdates(any())(any(), any())).
+          thenReturn(Future(IncomeSourcesObligationsTestConstants.testObligationsModel))
+
+        val result = TestUKPropertyAddedController.show(testSelfEmploymentId)(fakeRequestWithActiveSession.withSession(addUkPropertyStartDate -> "2022-01-01", addUkPropertyAccountingMethod -> "cash"))
+        status(result) shouldBe OK
+      }
+    }
+    "return 303 SEE_OTHER" when {
+      "Income Sources FS is disabled" in {
+        disable(IncomeSources)
+        setupMockAuthRetrievalSuccess(BaseTestConstants.testIndividualAuthSuccessWithSaUtrResponse())
+        mockUKPropertyIncomeSource()
+
+        val result = TestUKPropertyAddedController.show(testPropertyIncomeId)(fakeRequestWithActiveSession)
+        status(result) shouldBe SEE_OTHER
```

Comment:
Is redirecting to the homepage expected?

Reply:
When the FS is disabled yes